added initial set of descriptions for VM and VMSS properties

gatneil opened the pull request.

gatneil committed:
added initial set of descriptions for VM and VMSS properties

gatneil committed:
fixed descriptions

gatneil committed:
changed back location fields

gatneil committed:
reverted resource descriptions

veronicagg commented on specification/compute/resource-manager/Microsoft.Compute/2017-03-30/compute.json:
```diff
@@ -3413,11 +3413,11 @@
         },
         "type": {
           "type": "string",
```
based on the description you may want to evaluate turning this property into an enum in the future - https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md#x-ms-enum

gatneil replied:
Unfortunately the list of extension types is not static or versioned (at least, not in any user-specifiable way); so I don't think an enum would work for it (I *wish* it were versioned :))

veronicagg commented on specification/compute/resource-manager/Microsoft.Compute/2017-03-30/compute.json:
```diff
@@ -4893,7 +4893,7 @@
         "type": {
           "type": "string",
-          "description": "The type of identity used for the virtual machine scale set. Currently, the only supported type is 'SystemAssigned', which implicitly creates an identity.",
+          "description": "The type of indentity used for the virtual machine. Currently, the only supported type is 'SystemAssigned', which implicitly creates an identity.",
```
typo: indentity -> identity?

gatneil replied:
lol; thanks

veronicagg commented on specification/compute/resource-manager/Microsoft.Compute/2017-03-30/compute.json:
```diff
@@ -4976,14 +4976,14 @@
           "type": "string",
           "readOnly": true,
-          "description": "The type of restrictions.",
+          "description": "The type of restrictions",
```
"." at the end?

veronicagg commented on specification/compute/resource-manager/Microsoft.Compute/2017-03-30/compute.json:
```diff
@@ -5216,20 +5216,20 @@
         "createOption": {
           "$ref": "#/definitions/CreateOption",
-          "description": "The create option."
+          "description": "The create option"
```
"." at the end?

veronicagg commented on specification/compute/resource-manager/Microsoft.Compute/2017-03-30/compute.json:
```diff
@@ -5472,23 +5472,23 @@
         "type": {
           "type": "string",
-          "description": "The type of the extension handler."
+          "description": "Specifies the type of the extension; an example is \"CustomScriptExtension\""
```
"." at the end?

veronicagg commented on specification/compute/resource-manager/Microsoft.Compute/2017-03-30/compute.json:
```diff
@@ -5551,27 +5551,27 @@
         "diagnosticsProfile": {
           "$ref": "#/definitions/DiagnosticsProfile",
-          "description": "The virtual machine scale set diagnostics profile."
+          "description": "Specifies the boot diagnostic settings state. <br><br>Minimum api-version: 2015-06-15"
```
"." at the end?

veronicagg reviewed (changes requested):
Few minor comments, you may also want to evaluate adding missing descriptions flagged by linter tool, example: https://travis-ci.org/Azure/azure-rest-api-specs/jobs/267008359#L723, https://travis-ci.org/Azure/azure-rest-api-specs/jobs/267008359#L1695

gatneil committed:
fixed spelling and added missing periods

gatneil commented:
@veronicagg thanks for the review! Addressed the comments above. This particular PR is meant to get a jump start on descriptions, but look forward to more incremental follow-up PRs over time to bring it up to 100% description coverage :)

gatneil committed:
added missing periods

AutorestCI commented:
[![Build Status](https://travis-ci.org/AutorestCI/azure-sdk-for-ruby.svg?branch=RestAPI-PR1569)](https://travis-ci.org/AutorestCI/azure-sdk-for-ruby) https://github.com/Azure/azure-sdk-for-ruby/pull/922

AutorestCI commented:
No modification for AutorestCI/azure-sdk-for-node

AutorestCI commented:
[![Build Status](https://travis-ci.org/AutorestCI/azure-sdk-for-python.svg?branch=RestAPI-PR1569)](https://travis-ci.org/AutorestCI/azure-sdk-for-python) https://github.com/Azure/azure-sdk-for-python/pull/1388
